util/symbolic_trace/symcctrace/parsing.py

```diff
@@ -0,0 +1,128 @@
+import functools
+import itertools
+import json
+import math
+import pathlib
+
+import cattrs
+
+from . import data
+
+MEMORY_AREA_MAX_DISTANCE = 0x1000000
+
+
+def parse_trace(
+        trace_file: pathlib.Path,
+        memory_region_names_file: pathlib.Path,
+) -> data.TraceData:
+    """Parses a trace generated by the tracing feature of SymCC.
+
+    Args
+        trace_file: The path to the trace file generated by SymCC.
+        memory_region_names_file: The path to a file containing a mapping from memory region names to their addresses.
```

Comment:
How to generate such file using SymCC? What should be its format?

Reply:
I do not know how to generate  the memory region file with SymCC. I added code in SymQEMU such that it outputs this file, and I plan to submit a PR to SymQEMU with this feature.

I will document the format of the file, and I will also edit this python function such that the memory region files is not mandatory.

For a bit of context, the motivation for using this memory region information is that, because of ASLR (and because I was comparing the behavior of different versions of SymQEMU), equivalent memory areas were put at different locations in memory across different runs. To be able to compare different traces, I needed to express the addresses relatively to their area.

Reply:
So, you are adding SymQEMU specific code to the SymCC I don't think that this is a good idea, because not everyone who will use SymCC has to use SymQEMU. So I think that the solution should be made more general to work without SymQEMU. In you paper I see:
>  As a solution, we added some code in both versions of SymQEMU
that makes them output, in another json file, the addresses of the different memory regions,
including the addresses of the guest memory segments (stack, data, etc.)

Here I some my ideas how this can be improved:
- Maybe will be enough to use variables provided by linker? It provides you end of text, end of data and end of initialized memory symbols: https://stackoverflow.com/a/4309620.
- If you need more data you can add a code which reads `/proc/<pid>/maps` file to get addresses of interesting sections in runtime.
- You can start tested program using `setarch -R` to disable adress randomization
- Instead of addresses you can generate ID's in other more generic way (e.g. hashes similar to qsym).